tox.ini

```diff
@@ -0,0 +1,11 @@
+[tox]
+envlist =
+    py{311, 310, 39}
+skipsdist = true
+
+[testenv]
+passenv =
+    FORCE_COLOR
+commands =
+    make test
+allowlist_externals = make
```

Comment:
Why relay on `make`, another automation tool (and one that will not work cross-platform, so the tests cannot be run locally through tox on Windows), instead of just calling `pytest` directly? It will still automatically pick up the same args so long as they are properly configured in the `pytest.ini` as they should be, rather than stuck in a Makefile. This just all seems rather odd, at least to me...

Also, this means that like we already are in the build job, the dependencies are installed twice—once in the outer env, which are not used, and once in the inner env via `make venv`, which are (AFAIK).

Reply:
It was to handle installing requirements, avoid repeating the pytest command, and make sure the `make` command is tested. But good point about Windows! Will change it.

Requirements are only installed once in this workflow, currently via `make test` -> `make venv`.

Reply:
> It was to handle installing requirements, avoid repeating the pytest command, and make sure the make command is tested. But good point about Windows! Will change it.

Gotcha. The other benefit to changing it is that it makes it easier to add additional arguments to control the output of the CI job, which should be `-v --cov-report term` at least per the discussion above.

Ideally we'd fully replace `make` with drop-in equivalent `tox` tasks, but that's an issue for another day.

> Requirements are only installed once in this workflow, currently via make test -> make venv.

Ah, that's only in the build job—I got mixed up, sorry.